pgcli/packages/parseutils/__init__.py

```diff
@@ -1,34 +1,37 @@
 import sqlparse
 
 
+BASE_KEYWORDS = ["drop", "shutdown", "delete", "truncate", "alter"]
+ALL_KEYWORDS = BASE_KEYWORDS + ["update"]
+
+
 def query_starts_with(formatted_sql, prefixes):
     """Check if the query starts with any item from *prefixes*."""
     prefixes = [prefix.lower() for prefix in prefixes]
     return bool(formatted_sql) and formatted_sql.split()[0] in prefixes
 
 
-def query_is_unconditional_update(formatted_sql):
-    """Check if the query starts with UPDATE and contains no WHERE."""
-    tokens = formatted_sql.split()
-    return bool(tokens) and tokens[0] == "update" and "where" not in tokens
-
-
-def query_is_simple_update(formatted_sql):
-    """Check if the query starts with UPDATE."""
-    tokens = formatted_sql.split()
-    return bool(tokens) and tokens[0] == "update"
-
-
-def is_destructive(queries, warning_level="all"):
+def is_destructive(queries, keywords):
```

Comment:
Looks like we're losing a valuable behavior here. User can no longer differentiate between filtered update and unconditional update.

I see a couple ways to resolve this. We could have a special value in the list of keywords (for example, `unconditional_update`, that would warn on updates but only unconditional ones. That would have to be clearly commented in `pgclirc`.

Another way to solve this, if the user-provided list of destructive keywords is not empty, but does NOT contain keyword `update`, we could still force the warning on unconditional updates.

What do you think?

Reply:
Good point. I think `unconditional_update` makes sense.